src/jennifer/migration/table_builder/create_table.cr

```diff
@@ -93,7 +93,7 @@ module Jennifer
           column = Inflector.foreign_key(name)
           is_null = options.has_key?(:null) ? options[:null] : true
 
-          integer(column, { :type => type, :null => is_null })
+          field(column, type, { :null => is_null })
```

Comment:
in common case you would like to use "internal types". To achieve this and also allow developer to pass native type if required suggest 

```crystal
field_internal_type = options.has_key?(:sql_type) ? nil : type
@fields[name.to_s] = build_column_options(field_internal_type, options)
```

Reply:
Yes. I modified it.